internal/operands/common-templates/reconcile.go

```diff
@@ -122,6 +130,53 @@ func reconcileEditRole(request *common.Request) (common.ResourceStatus, error) {
 			foundRole.Rules = newRole.Rules
 		})
 }
+func updateOldTemplates(request *common.Request) (common.ResourceStatus, error) {
+	baseRequirement, err := labels.NewRequirement("template.kubevirt.io/type", selection.Equals, []string{"base"})
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during preparing of type label requirement, %v", err))
+		panic(err)
+	}
+	var versionRequirement *labels.Requirement
+	versionRequirement, err = labels.NewRequirement("template.kubevirt.io/version", selection.NotEquals, []string{Version})
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during preparing of version label requirement, %v", err))
+		panic(err)
+	}
+
+	labelsSelector := labels.NewSelector().Add(*baseRequirement, *versionRequirement)
+
+	opts := client.ListOptions{
+		LabelSelector: labelsSelector,
+	}
+
+	var oldTemplates templatev1.TemplateList
+	err = request.Client.List(request.Context, &oldTemplates, &opts)
+	if err != nil {
+		request.Logger.Error(err, fmt.Sprintf("Error during listing templates, %v", err))
+		panic(err)
+	}
+
+	updatedTemplates := make([]templatev1.Template, 0)
+	for _, template := range oldTemplates.Items {
+		for key := range template.Labels {
+			if strings.HasPrefix(key, "os.template.kubevirt.io/") || strings.HasPrefix(key, "flavor.template.kubevirt.io/") || strings.HasPrefix(key, "workload.template.kubevirt.io/") {
+				delete(template.Labels, key)
+			}
+		}
+		updatedTemplates = append(updatedTemplates, template)
+	}
+	oldTemplates.Items = updatedTemplates
+
+	for _, template := range oldTemplates.Items {
+		err = request.Client.Update(request.Context, &template)
+		if err != nil {
+			request.Logger.Error(err, fmt.Sprintf("Error during updating of templates, %v", err))
+			panic(err)
+		}
+	}
+
+	return common.ResourceStatus{}, nil
```

Comment:
@akrejcir, @vatsalparekh what should be returned in this case?

Reply:
For other resources, this return value signals if the resource is still being updated, or if it is in the desired state. If any resource is still updating, the operator will not be in phase `deployed`.

In this case however,  the operator does not own any of the old templates, so I think they should not block the operator from being in `deployed` state. The return value can be left as is now.